src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -43,11 +44,12 @@ public static IEnumerable<IPackage> GetPackages(ChocolateyConfiguration configur
                 results = results.Where(p => p.IsLatestVersion);
             }
 
-            return results.OrderBy(p => p.Id)
-                          .AsEnumerable()
-                          .Where(PackageExtensions.IsListed)
-                          .Where(p => configuration.Prerelease || p.IsReleaseVersion())
-                          .distinct_last(PackageEqualityComparer.Id, PackageComparer.Version).ToList();
+            // just trust the server order, don't sort because that's a blocking operation
+            // also don't worry about multiple versions, considering Is*LatestVersion only applies to one

```

Comment:
NuGet Server _currently_ uses OData under the covers, but it's only one possible provider. The FileSystem implementation doesn't, for instance (although it's presumably so much faster nobody cares).  Anyway. Like I said, totally up to you.